### ISSUE TITLE
issue(gcp): metrics - klibs(gcp,tls) - memory leak (maybe)

### ISSUE THREAD

**Issue by rinor** [nanovms/nanos#1912]
I've been testing/using `gcp metrics klib` functionality on some services (not important here, but implemented in different languages rust, go, typescript) and have observed some weird memory behavior/increase reported/shown on `OBSERVABILITY/Metrics` of gcp dashboard. 
Even though the services were mostly idle, the user space memory kept increasing steadily until it reached ~100% memory usage and at some point before that,  `gcp metrics klib` stops reporting data, although there are no errors shown on console logs. It was weird since some of those services deployed on linux use around 100 MiB on avg (ofc depends on traffic/usage/other) .

Being a bit confused, today decided to deploy an instance, without public network access, that does nothing (almost).

<details><summary><code>the instance was started today `2023-08-06` at `08:15` and the daily report of ~24 hours</code></summary>

| Time (approx)  | Mem usage |
| :-------------------: | :---------------: |
| 08:15 | 66.85 MiB
| 10:15 | 67.25 MiB 
| 10:20 | 68.85 MiB 
| 12:20 | 69.65 MiB 
| 12:25 | 70.85 MiB 
| 14:25 | 72.05 MiB 
| 14:30 | 72.85 MiB 
| 16:30 | 74.85 MiB
| 18:30 | 76.85 MiB
| 20:35 | 77.35 MiB
| 20:40 | 78.85 MiB
| 22:40 | 79.65 MiB
| 22:45 | 80.85 MiB
| 00:45 | 82.05 MiB
| 00:50 | 82.85 MiB
| 02:50 | 84.05 MiB
| 02:55 | 84.85 MiB
| 04:55 | 86.45 MiB
| 05:00 | 86.85 MiB
| 07:00 | 88.85 MiB
| 09:00 | 90.85 MiB
</details>

<details><summary><b>code</b><code>pause.c</code>file</summary>

```sh
$ gcc pause.c -o pausec -s
```

```c
#include <unistd.h>

int main() {
        while(1) pause();
}
```
</details>

<details><summary><b>ops</b><code>config.json</code>file</summary>

```json
{
  "Klibs": [
    "gcp",
    "tls"
  ],
  "ManifestPassthrough": {
    "gcp": {
      "metrics": {
        "interval": "60"
      }
    }
  },
  "CloudConfig": {
    "Platform": "gcp",
    "ProjectID": "xxxx-xxxx",
    "BucketName": "yyyy-yyyy",
    "Flavor": "f1-micro",
    "InstanceProfile": "default",
    "VPC": "default",
    "Subnet": "",
    "Zone": "us-central1-a"
}
```
</details>

**Comment by francescolavra** [comment]
The memory leaks is actually not in the klibs, but in the kernel code that implements in-kernel TCP connections.
Will be fixed by #1919.

**Comment by rinor** [comment]
Just to confirm that everything works as expected. Running the fix for 48 hours now and no leak observed. Thank you.
